pkg/pod/pod_test.go

```diff
@@ -2006,6 +2006,147 @@ debug-fail-continue-heredoc-randomly-generated-mz4c7
 	}
 }
 
+func TestPodBuild_TaskLevelResourceRequirements(t *testing.T) {
```

Comment:
this looks great! could you also add a test for where the steps have resource requirements?